### DISCUSSION TITLE
Some dark scenes are very dark with DV

### DISCUSSION THREAD

**Discussion by Jerroder** [R3S3t9999/DoVi_Scripts#55] [General]
Hi,

This isn't directly related to any scripts but I feel like it's the right place given that you seem to be very knowledgeable about DV. I have a problem with Dolby Vision files which doesn't happen with SDR or HDR10 content. I play Dolby vision from Jellyfin, with Infuse on an Apple TV 4k 1st gen connected to an LG OLED CX. 

At first I thought it was the infamous LG OLED dimming but even when disabling TPC and GSR in the service menu, it still happens. We can see the problem very clearly in [this scene](https://imgur.com/OSDkUrk) (the camera has ISO, aperture and shutter speed locked). Because it still happens with Auto Static Brightness Limiter off, I'm thinking it's a content issue? Although the [subtitles dim too](https://imgur.com/TOCxqFQ).

On my MacBook, still playing with Infuse, the scenes are still dark but the subtitles don't dim anymore, that leads me to think it's just a difference of how the TV and MacBook deal with dynamic metadata, and that it really is a metadata problem. Also, when disable "match range" in the Apple TV and just playing the file in SDR or HDR, the scenes aren't dark anymore.

When looking at files this happens with, I noticed they seem to be DV Profile 5. I don't have the problem with all P5, but I think every time the problem occurs, it's P5, and it never seem to happen with P8.1. Before I replace my whole P5 library with P8.1 and because P5 is supposed to be higher quality, is there something else I should look into? I don't know much about it, can it be a problem with MEL because some files don't have FEL?

Edit: just tested on a non-DV player, with Jellyfin hardware transcoding using Intel QSV VPP Tone Mapping and the scenes look fine, so Infuse doesn't handle metadata correctly but the Intel Hardware does?

**Comment by R3S3t9999** [comment]
Check the L1-L2 metadata of that scene. 
It could be a case of badly generated/made metadata because when the DV metadata is within your TV capabilities,  it just tells the TV not to apply any tone mapping and display the content as it is. [You can test these dark scenes too.](https://drive.google.com/drive/u/1/folders/1OH-DT9jfodpGKX9LwN5FsewWGPBII1Ib)

You probably already know that but to check a specific frame for the metadata, input the RPU in workflow 1 and select ''e''. It will ask you a frame number and display the metadata like this.

      "ext_metadata_blocks": [
        {
          "Level1": {
            "min_pq": 0,
            "max_pq": 2081,
            "avg_pq": 1229
          }
        },
        {
          "Level2": {
            "target_max_pq": 2081,
            "trim_slope": 2008,
            "trim_offset": 2016,
            "trim_power": 1333,
            "trim_chroma_weight": 2048,
            "trim_saturation_gain": 2048,
            "ms_weight": 1642
          }
        },
        {
          "Level2": {
            "target_max_pq": 2851,
            "trim_slope": 2059,
            "trim_offset": 2048,
            "trim_power": 1955,
            "trim_chroma_weight": 2048,
            "trim_saturation_gain": 2048,
            "ms_weight": 1956
          }
        },
        {
          "Level2": {
            "target_max_pq": 3079,
            "trim_slope": 2049,
            "trim_offset": 2048,
            "trim_power": 2047,
            "trim_chroma_weight": 2048,
            "trim_saturation_gain": 2048,
            "ms_weight": 2048
          }
        },
        {
          "Level5": {
            "active_area_left_offset": 0,
            "active_area_right_offset": 0,
            "active_area_top_offset": 0,
            "active_area_bottom_offset": 0
          }
        },
        {
          "Level6": {
            "max_display_mastering_luminance": 1000,
            "min_display_mastering_luminance": 1,
            "max_content_light_level": 0,
            "max_frame_average_light_level": 0
          }
        }
      ]
    },
    "cmv40_metadata": {
      "num_ext_blocks": 5,
      "ext_metadata_blocks": [
        {
          "Level3": {
            "min_pq_offset": 2048,
            "max_pq_offset": 1547,
            "avg_pq_offset": 1435
          }
        },
        {
          "Level8": {
            "length": 13,
            "target_display_index": 1,
            "trim_slope": 2048,
            "trim_offset": 2048,
            "trim_power": 2048,
            "trim_chroma_weight": 2048,
            "trim_saturation_gain": 2048,
            "ms_weight": 1642,
            "target_mid_contrast": 2048,
            "clip_trim": 2075
          }
        },

**Reply by Jerroder** [reply]
> [You can test these dark scenes too.](https://drive.google.com/drive/u/1/folders/1OH-DT9jfodpGKX9LwN5FsewWGPBII1Ib)

Those are very dark on my MacBook, especially Dune and Batman, I'll check on the TV when I get home.

> You probably already know that

I actually have no idea what you're talking about, what's workflow 1 and how do I get the RPU?

**Comment by R3S3t9999** [comment]
> Those are very dark on my MacBook, especially Dune and Batman, I'll check on the TV when I get home.

Yes, they are supposed to be dark. The point is to compare it with HDR10. On my LG C2, they look identical in DV/HDR10 but they do have proper DV metadata.


> I actually have no idea what you're talking about, what's workflow 1 and how do I get the RPU?

Well, you're asking on my DoVi_Scripts github so I assumed you were already using it. Check out the tutorial installation. Extract your RPU file with workflow 1 (select ''e'') and then run the RPU in workflow 1 again, select ''e'' and specify the frame number you want to check.

**Reply by Jerroder** [reply]
If I didn't mess up, this is the metadata for one of those frame:

```
    "cmv29_metadata": {
      "num_ext_blocks": 5,
      "ext_metadata_blocks": [
        {
          "Level1": {
            "min_pq": 0,
            "max_pq": 3493,
            "avg_pq": 2048
          }
        },
        {
          "Level2": {
            "target_max_pq": 2081,
            "trim_slope": 1562,
            "trim_offset": 2052,
            "trim_power": 1413,
            "trim_chroma_weight": 2048,
            "trim_saturation_gain": 1638,
            "ms_weight": 2048
          }
        },
        {
          "Level2": {
            "target_max_pq": 2851,
            "trim_slope": 2083,
            "trim_offset": 2068,
            "trim_power": 2232,
            "trim_chroma_weight": 2048,
            "trim_saturation_gain": 2005,
            "ms_weight": 2048
          }
        },
        {
          "Level2": {
            "target_max_pq": 3079,
            "trim_slope": 2137,
            "trim_offset": 2052,
            "trim_power": 2404,
            "trim_chroma_weight": 2048,
            "trim_saturation_gain": 2048,
            "ms_weight": 2048
          }
        },
        {
          "Level6": {
            "max_display_mastering_luminance": 1000,
            "min_display_mastering_luminance": 1,
            "max_content_light_level": 0,
            "max_frame_average_light_level": 0
          }
        }
      ]
    }
  }
}
```

I want to add that I tried to play it from a non-DV compatible device, so the video was transcoded and tone mapped on the fly, and I got an error from ffmpeg on Jellyfin saying "Multiple Dolby Vision RPUs found in one AU.". Can this be a problem? Like the file has been badly done and the TV plays a bad RPU?

**Comment by R3S3t9999** [comment]
these metadata will certainly darken the image if you watch on a low nits display but that's normal if the scene really is like that. 
Which movie and frame is it?

max_pq is at 2530nits and the 600/1000 nits trims darken quite a bit with the ''L2 power'' metadata

**Reply by Jerroder** [reply]
This is Ted Lasso S03E11 frame 23472, but I don't think an LG OLED CX is considered a "low nits display". You can see [the last scene change](https://imgur.com/OSDkUrk) is much darker than the beginning, I doubt it's intended.

**Comment by R3S3t9999** [comment]
it is definitely a case of bad DV metadata. That frame has very small 600nits highlights but the DV metadata tells the display to dim like they were 2530nits highlights + even more diming with the L2 trims.

I don't know about all the season episodes, but for this whole episode, the DV metadata makes no sense. Here's a comparison between the HDR measurement and the DV metadata. https://slow.pics/c/GbjFZCm2

**Reply by Jerroder** [reply]
> it is definitely a case of bad DV metadata.

It's weird that hardware tone mapping doesn't have the problem even with bad metadata. Is this the only way to check for this? What should I look at exactly? Is this:

```
L6 metadata: Mastering display: 0.0001/1000 nits. MaxCLL: 0 nits, MaxFALL: 0 nits
L2 trims: 100 nits, 600 nits, 1000 nits
```
and
```
"max_pq": 3493,
```
?

**Comment by R3S3t9999** [comment]
Yes max_pq and the L2 trims are not right for this whole episode. L6 doesn't matter.
I did a sample of that frame with Dolby Vision metadata and no metadata, they did a horrible job with DV:
https://drive.google.com/drive/folders/1100OWrD7m9K9V4etEEpxILImias9GGKR?usp=drive_link

**Reply by Jerroder** [reply]
Ah, I feel better knowing it's not a TV problem, even though this happens with many files.

But just so I understand, the problem here is that L2 trims is mastered for 600 (or 1000?) nits, but max_pq is 3493? What should good values be, higher nits or lower max_pq?

**Comment by R3S3t9999** [comment]
the problem is the values are not suited for the video's actual brightness so for example in that shot, the TV thinks it has to tone map a 2500nits image to 800nits.  It's normal to have 100-600-1000nits L2 trim and proper metadata for this shot would look like this (684nits). If all the episodes are like that, you should watch it in HDR10.

        {
          "Level1": {
            "min_pq": 0,
            "max_pq": 2910,
            "avg_pq": 1439
          }
        },
        {
          "Level2": {
            "target_max_pq": 2081,
            "trim_slope": 1698,
            "trim_offset": 2028,
            "trim_power": 1521,
            "trim_chroma_weight": 2048,
            "trim_saturation_gain": 2048,
            "ms_weight": 2048
          }
        },
        {
          "Level2": {
            "target_max_pq": 2851,
            "trim_slope": 1988,
            "trim_offset": 2047,
            "trim_power": 1886,
            "trim_chroma_weight": 2048,
            "trim_saturation_gain": 2048,
            "ms_weight": 2048
          }
        },
        {
          "Level2": {
            "target_max_pq": 3079,
            "trim_slope": 2049,
            "trim_offset": 2048,
            "trim_power": 2046,
            "trim_chroma_weight": 2048,
            "trim_saturation_gain": 2048,
            "ms_weight": 2048
          }
        },
        {
          "Level5": {
            "active_area_left_offset": 0,
            "active_area_right_offset": 0,
            "active_area_top_offset": 0,
            "active_area_bottom_offset": 0
          }

**Reply by Jerroder** [reply]
> a 2500nits image to 800nits.

> proper metadata for this shot would look like this (684nits).

I don't know what to look at to come to this conclusion, how do you get 2500, 800 and 684 from these metadata? What makes your metadata more correct than mine?

**Comment by R3S3t9999** [comment]
this shot actual measured brightness is 684nits... 
The DV L1 maxpq metadata is 2530nits... 
Your TV is capable of about 800nits.

The Dolby Vision metadata are in 12bit PQ which can be converted to nits. Workflow 9-8 in my [scripts.](https://github.com/R3S3t9999/DoVi_Scripts)

![WindowsTerminal_eutylBeRno](https://github.com/user-attachments/assets/bb6a49b9-aa68-4fc9-bac8-b60a2e8823bb)

**Reply by Jerroder** [reply]
Alright, I understand.

One last question, where did you get the correct metadata (2910 max_pq)? Is it from another file or derived from HDR10?

**Comment by R3S3t9999** [comment]
I generated proper metadata with the [Dolby Vision algorithm](https://www.youtube.com/watch?v=FVSh3oGqfXY).

**Reply by Jerroder** [reply]
Thank you very much for your help and information!
